docs/proposals/20210624-cloneset-scaledown-topology-spread.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Proposal Template
+authors:
+- "@FillZpp"
+  reviewers:
+- "@Fei-Guo"
+- "@furykerry"
+  creation-date: 2021-06-24
+  last-updated: 2021-06-24
+  status: implementable
+---
+
+# CloneSet scale-down supports topology spread
+
+## Motivation
+
+Currently, the sequence of CloneSet scale-down is:
+
+1. Node unassigned < assigned
+2. PodPending < PodUnknown < PodRunning
+3. Not ready < ready
+4. Lower pod-deletion cost < higher pod-deletion-cost
+5. Been ready for empty time < less time < more time
+6. Pods with containers with higher restart counts < lower restart counts
+7. Empty creation time pods < newer pods < older pods
+
+What's more, CloneSet should also choose pods to delete according to topology constraints.
+
+## User Stories
+
+1. Pods stacking on the same node should be deleted preferentially.
+2. Should delete Pods according to topology spread constraints.
+
+## Implementation Details/Notes/Constraints
+
+Add a rank step between (4) and (5). We can now support two Ranker types:
```

Comment:
please describe how to sort the pods using both the SameNodeRanker and SpreadConstraintsRanker